storage/stored_transaction.go

```diff
@@ -209,40 +202,48 @@ func (s *Storage) initBatchedTxsCounter() (err error) {
 	return err
 }
 
-func (s *TransactionStorage) GetTransactionHashesByBatchIDs(batchIDs ...models.Uint256) ([]common.Hash, error) {
-	hashes := make([]common.Hash, 0, len(batchIDs)*32)
-
-	// We have an index on CommitmentID. It turns out that BatchID is the first
-	// member of the BatchID struct, so we effectively have an index on BatchID.
-	// We can take advantage of that index by manually iterating.
-	// The slow version is: Badger.Find(..., bh.Where("CommitmentID.BatchID").In(...))
-
-	var keyList bh.KeyList
-	batchPrefixes := batchIdsToBatchPrefixes(batchIDs)
-	seekPrefix := db.IndexKeyPrefix(stored.BatchedTxName, "CommitmentID")
-	err := s.database.Badger.Iterator(seekPrefix, db.ReversePrefetchIteratorOpts, func(item *bdg.Item) (bool, error) {
-		if validForPrefixes(keyValue(seekPrefix, item.Key()), batchPrefixes) {
-			err := item.Value(func(val []byte) error {
-				return db.Decode(val, &keyList)
-			})
-			if err != nil {
-				return false, err
-			}
-			txHashes, err := decodeKeyListHashes(stored.BatchedTxPrefix, keyList)
-			if err != nil {
-				return false, err
-			}
-			hashes = append(hashes, txHashes...)
+func (s *TransactionStorage) getTransactionSlotsByBatchID(batchID models.Uint256) ([]models.CommitmentSlot, error) {
+	slots := make([]models.CommitmentSlot, 0, 32)
+
+	// nolint: gocritic
+	seekPrefix := append(stored.BatchedTxPrefix, batchID.Bytes()...)
+
+	// BatchedTx are stored with CommitmentSlot as their primary key: BatchID is the
+	// first member of CommitmentSlot which means we effectively have an index on
+	// BatchID.
+
+	var batchedTx stored.BatchedTx
+	err := s.database.Badger.Iterator(seekPrefix, db.PrefetchIteratorOpts, func(item *bdg.Item) (bool, error) {
+		err := item.Value(batchedTx.SetBytes)
```

Comment:
You can make this more efficient by disabling prefetch (use `db.KeyIteratorOpts`) and extracting the `CommitmentSlot` from `item.Key()`

Reply:
good point!